### DISCUSSION TITLE
New Mod menu Keybinds should be able to cycle between variant mods

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#18417] [Ideas]
This is essentially what I mean:
![image](https://user-images.githubusercontent.com/64581009/170359151-bf610528-d16b-4ed7-8d20-16913132aae4.png)

The number of times I've turned on Perfect when I wanted to select Double Time is just quite annoying. The same goes for Hidden too.
I personally think it's better to cycle through the variants mods with one button rather making every mod separate. Less space and compact too.

Even better if you can bind mods to one key and able to cycle through all of them with that one key.

_Edit: I've just noticed that this idea is exactly the same behavior as stable but you get the idea._

**Comment by bdach** [comment]
the issue is how to make it understandable to the user which letter is which mod?

**Reply by NotGumballer91** [reply]
Unfortunately, I really can't think of anything for that. This function is only intuitive for the old design.
The best I got would be to show all the mods in the key configurations. But other rulesets might make it more complicated.

**Reply by evilldev** [reply]
I think it would be a good idea to add a small "use stable keybinds" button somewhere

**Reply by NotGumballer91** [reply]
No.

**Comment by zachmanthethird** [comment]
Personally, the only qualm I’ve had with the new keybinds is the fact that they aren’t universal across all rulesets anymore. Currently, this only manifests as Hidden and Flashlight being different for mania; however, adding keybinds for the Conversion and Fun sections will promote similar issues. Having different rulesets display the same mods in other locations is incredibly unintuitive.

That being said, there isn’t a straightforward solution to this problem. I would prefer that the four base rulesets all have mods aligned to the same locations, but this can’t and shouldn’t be extended to custom rulesets.
